x-pack/platform/plugins/shared/alerting/public/pages/maintenance_windows/components/maintenance_window_solution_selection.test.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { screen, fireEvent, waitFor } from '@testing-library/react';
+import { MaintenanceWindowSolutionSelection } from './maintenance_window_solution_selection';
+import type { AppMockRenderer } from '../../../lib/test_utils';
+import { createAppMockRenderer } from '../../../lib/test_utils';
+
+const mockOnChange = jest.fn();
+
+describe('maintenanceWindowCategorySelection', () => {
+  let appMockRenderer: AppMockRenderer;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    appMockRenderer = createAppMockRenderer();
+  });
+
+  it('should not display radio group if scoped query is disabled', async () => {
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        selectedSolution={''}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+        isScopedQueryEnabled={false}
+      />
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.queryByTestId('maintenanceWindowSolutionSelectionLoading')
+      ).not.toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      expect(screen.queryByTestId('maintenanceWindowSolutionSelection')).not.toBeInTheDocument();
+    });
+  });
+
+  it('should display radio group if scoped query is enabled', async () => {
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        selectedSolution={''}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+        isScopedQueryEnabled={true}
+      />
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.queryByTestId('maintenanceWindowSolutionSelectionLoading')
+      ).not.toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      expect(screen.queryByTestId('maintenanceWindowSolutionSelection')).toBeInTheDocument();
```

Comment:
nit: `queryByTestId` [should only be used for querying for non-existence](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence).